adapted-video-frontend/src/main/java/fr/labri/progress/comet/service/SwiftServiceImp.java

```diff
@@ -0,0 +1,203 @@
+			throw new WebApplicationException(response.getStatus());
+
+		case BAD_REQUEST:
+			LOGGER.error("Bad URL for swift");
+			throw new WebApplicationException(response.getStatus());
+
+		default:
+			LOGGER.error("Swift connection probleme: {}", response.getStatus());
+			throw new WebApplicationException(response.getStatus());
+		}
+//		try {
+//			xStorageUrl = new URL(response.getHeaderString("X-Storage-Url"));
+//		} catch (MalformedURLException e1) {
+//			// TODO Auto-generated catch block
+//			e1.printStackTrace();
+//		}
+		
+		
+		
+		target = client.target(xStorageUrl.toString()+"/"+id);
+		LOGGER.debug("Send request to server  {}", target.getUri());
+		response = target.request().header("X-Auth-Token", xAuthToken)
+				.put(Entity.json("hello"), Response.class);
+
+		switch (Status.fromStatusCode(response.getStatus())) {
+		// TODO: verfy this return message
+		case ACCEPTED:
+		case CREATED:
+			LOGGER.debug("connetion to swift ok");
+
+			break;
+		case UNAUTHORIZED:
+			LOGGER.error("Bad USERNAME or PASSWORD for swift");
+			throw new WebApplicationException(response.getStatus());
+
+		case BAD_REQUEST:
+			LOGGER.error("Bad URL for swift");
+			throw new WebApplicationException(response.getStatus());
+
+		default:
+			LOGGER.error("Swift connection probleme: {}", response.getStatus());
+			throw new WebApplicationException(response.getStatus());
+		}
+//		try {
+//			xStorageUrl = new URL(response.getHeaderString("X-Storage-Url"));
+//		} catch (MalformedURLException e1) {
+//			// TODO Auto-generated catch block
+//			e1.printStackTrace();
+//		}
+
+
+	}
+	
+
+
+	@Override
+	public URL GenerateReturnURI(String name, String id) {
+		return GenerateReturnURI( name,  id,"PUT");
+	}
+	
+
+	@Override
+	public URL GenerateReturnURI(String name, String id,String methode) {
+
+
+		// @debug:on
+		if (xStorageUrl == null) {
+			LOGGER.warn("Debug mode");
+			try {
+				xStorageUrl = new URL("http://swift-proxy.fr:8956/v1/auth_admin");
+			} catch (MalformedURLException e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
+			}
+		}
+		// @debug:off
+
+		long duration_in_seconds = 60 * 60 * 4;// 4 hours
+		long expire = duration_in_seconds + (System.currentTimeMillis() / 1000L);
+//		long expire = 1458746829L;
+		String path = xStorageUrl.getPath() + "/" + id + "/" + name;
+		String hmac_body = methode + "\n" + expire + "\n" + path;
+
+		SecretKeySpec keySpec = new SecretKeySpec(sharedKey.getBytes(), "HmacSHA1");
+
+		try {
+			Mac mac = Mac.getInstance("HmacSHA1");
+			mac.init(keySpec);
+			byte[] rawHmac = mac.doFinal(hmac_body.getBytes());
+			String result = BaseEncoding.base16().lowerCase().encode(rawHmac);
+			String returnUrl = xStorageUrl.getProtocol() + "://" + xStorageUrl.getAuthority() + path + "?temp_url_sig="
+					+ result + "&temp_url_expires=" + expire;
+
+			return new URL(returnUrl);
+		} catch (InvalidKeyException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
you should indicate what are the impacts of this error (and log it) not just print the stacktrace